### ISSUE TITLE
The shape of aud_features ?

### ISSUE THREAD

**Issue by kike-0304** [ZiqiaoPeng/SyncTalk#14]
`first_frame, last_frame = outputs[:1], outputs[-1:]
 aud_features = torch.cat([first_frame.repeat(2, 1), outputs, last_frame.repeat(2, 1)], dim=0).numpy()`
Why copy a few copies before and after extracting audio features？Do I also need to process aud_features in this way during training ?

**Comment by ZiqiaoPeng** [comment]
Since audio features are extracted by taking one feature per time segment, the first and last frames might be lost. The training data provided has already been processed.

**Comment by johndpope** [comment]
just a stab in the dark - a different paper - EMO - grab a few frames too.
https://arxiv.org/html/2402.17485v1

<img width="1432" alt="Screenshot 2024-03-06 at 9 46 27 am" src="https://github.com/ZiqiaoPeng/SyncTalk/assets/289994/f00ac493-293c-4346-ae39-f1931d87c4e2">

**Comment by ZiqiaoPeng** [comment]
You can change it to
`
first_frame, last_frame = outputs[:2], outputs[-2:] 
aud_features = torch.cat([first_frame, outputs, last_frame], dim=0).numpy()
`
But I tested it, and it didn't make much difference to the results.

**Comment by Elysiameik** [comment]
> You can change it to `first_frame, last_frame = outputs[:2], outputs[-2:] aud_features = torch.cat([first_frame, outputs, last_frame], dim=0).numpy()` But I tested it, and it didn't make much difference to the results.

 if self.opt.asr_model == 'ave':
                    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
                    model = AudioEncoder().to(device).eval()
                    ckpt = torch.load('./nerf_triplane/checkpoints/audio_visual_encoder.pth')
                    model.load_state_dict({f'audio_encoder.{k}': v for k, v in ckpt.items()})
                    dataset = AudDataset(self.opt.aud)
                    data_loader = DataLoader(dataset, batch_size=64, shuffle=False)
                    outputs = []
                    for mel in data_loader:
                        mel = mel.to(device)
                        with torch.no_grad():
                            out = model(mel)
                        outputs.append(out)
                    outputs = torch.cat(outputs, dim=0).cpu()
                    first_frame, last_frame = outputs[:1], outputs[-1:]
                    aud_features = torch.cat([first_frame.repeat(2, 1), outputs, last_frame.repeat(2, 1)], dim=0).numpy()
                else:
                    try:
                        aud_features = np.load(self.opt.aud)                                                             

I use this code to generate aud_ave.npy and use it to replace aud_ave.npy in May. The rest data of May remains unchanged. The result of training is that the mouth does not open. Is it wrong to generate aud_ave.npy like this? How do I get aud_ave.npy correctly?

**Comment by ZiqiaoPeng** [comment]
The code to get aud_ave.npy has been released.

**Comment by piwawa** [comment]
> The code to get aud_ave.npy has been released.

![image](https://github.com/ZiqiaoPeng/SyncTalk/assets/46926496/2b35874b-d4dd-49cd-9760-8e419fa49258)

Where is the `aud_ave.npy`? I have completed training and inference processes.

**Comment by ytm-01** [comment]
> 用于获取 aud_ave.npy 的代码已发布。

在哪里
